### PR TITLE
feat: upgrade chainlink keeper to v1.2

### LINKED ISSUE #891: use chainlink keeper registry v1.2 in next migration script
this automatically accepts registrations, instead of waiting for manual verification on their side

guide here https://docs.chain.link/docs/chainlink-automation/register-upkeep/

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
closes #891

tested successfully onfork with pending pr #961, which now gives a upkeep registered event instead of a registration pending.

old situation:
```
Events In This Transaction
--------------------------
├── ChainLink Token (0x514910771AF9Ca656af840dff83E8264EcF986CA)
│   ├── Transfer
│   │   ├── from: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│   │   ├── to: 0x09B1de9Ecf6600FD44e1cB1b46DF6B520f33ec6d
│   │   └── value: 80000000000000000000
│   └── Transfer
│       ├── from: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│       ├── to: 0x09B1de9Ecf6600FD44e1cB1b46DF6B520f33ec6d
│       ├── value: 80000000000000000000
│       └── data: 0xc4110e5c00000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000140000000000000000000000000c485afd2f3252ccb69d1c94392701d51013d42eb00000000000000000000000000000000000000000000000000000000000afc8000000000000000000000000086cbd0ce0c087b482782c181da8d191de18c82750000000000000000000000000000000000000000000000000000000000000180000000000000000000000000000000000000000000000004563918244f400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000135472656173757279566f7465724d6f64756c65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
│   
├── IUpkeepRegistrationRequests (0x09B1de9Ecf6600FD44e1cB1b46DF6B520f33ec6d)
│   └── RegistrationRequested
│       ├── hash: 0xe12da4a89762f3803cb68c9b6658ab59d77e69af94a8417729a384aed91bb5ad
│       ├── name: TreasuryVoterModule
│       ├── encryptedEmail: 0x00
│       ├── upkeepContract: 0xC485aFD2f3252cCB69d1c94392701d51013D42eB
│       ├── gasLimit: 720000
│       ├── adminAddress: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│       ├── checkData: 0x00
│       ├── amount: 80000000000000000000
│       └── source: 0
│   
└── Gnosis Safe (0x86cbD0ce0c087b482782c181dA8d191De18C8275)
    └── ExecutionSuccess
        ├── txHash: 0x95675f29636a14728e2462c089bcb8fea3961550e1ee6c2571ca400b9878e798
        └── payment: 0
```

new:
```
Events In This Transaction
--------------------------
├── ChainLink Token (0x514910771AF9Ca656af840dff83E8264EcF986CA)
│   ├── Transfer
│   │   ├── from: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│   │   ├── to: 0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d
│   │   └── value: 80000000000000000000
│   └── Transfer
│       ├── from: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│       ├── to: 0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d
│       ├── value: 80000000000000000000
│       └── data: 0x3659d66600000000000000000000000000000000000000000000000000000000000001200000000000000000000000000000000000000000000000000000000000000160000000000000000000000000c485afd2f3252ccb69d1c94392701d51013d42eb00000000000000000000000000000000000000000000000000000000000afc8000000000000000000000000086cbd0ce0c087b482782c181da8d191de18c827500000000000000000000000000000000000000000000000000000000000001a0000000000000000000000000000000000000000000000004563918244f400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000086cbd0ce0c087b482782c181da8d191de18c827500000000000000000000000000000000000000000000000000000000000000135472656173757279566f7465724d6f64756c65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
│   
├── IKeeperRegistrar (0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d)
│   └── RegistrationRequested
│       ├── hash: 0xe12da4a89762f3803cb68c9b6658ab59d77e69af94a8417729a384aed91bb5ad
│       ├── name: TreasuryVoterModule
│       ├── encryptedEmail: 0x00
│       ├── upkeepContract: 0xC485aFD2f3252cCB69d1c94392701d51013D42eB
│       ├── gasLimit: 720000
│       ├── adminAddress: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│       ├── checkData: 0x00
│       ├── amount: 80000000000000000000
│       └── source: 0
│   
├── IKeeperRegistry (0x02777053d6764996e594c3E88AF1D58D5363a2e6)
│   └── UpkeepRegistered
│       ├── id: 76213945328454238082937457532622255380921944950392826781659763134532367343725
│       ├── executeGas: 720000
│       └── admin: 0x86cbD0ce0c087b482782c181dA8d191De18C8275
│   
├── ChainLink Token (0x514910771AF9Ca656af840dff83E8264EcF986CA)
│   ├── Transfer
│   │   ├── from: 0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d
│   │   ├── to: 0x02777053d6764996e594c3E88AF1D58D5363a2e6
│   │   └── value: 80000000000000000000
│   └── Transfer
│       ├── from: 0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d
│       ├── to: 0x02777053d6764996e594c3E88AF1D58D5363a2e6
│       ├── value: 80000000000000000000
│       └── data: 0xa87f907bcdf883565ed3b0ea7b8e23c15c837c23200486e363e756a6ac22e06d
│   
├── IKeeperRegistry (0x02777053d6764996e594c3E88AF1D58D5363a2e6)
│   └── FundsAdded
│       ├── id: 76213945328454238082937457532622255380921944950392826781659763134532367343725
│       ├── from: 0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d
│       └── amount: 80000000000000000000
│   
├── IKeeperRegistrar (0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d)
│   └── RegistrationApproved
│       ├── hash: 0xe12da4a89762f3803cb68c9b6658ab59d77e69af94a8417729a384aed91bb5ad
│       ├── displayName: TreasuryVoterModule
│       └── upkeepId: 76213945328454238082937457532622255380921944950392826781659763134532367343725
│   
└── Gnosis Safe (0x86cbD0ce0c087b482782c181dA8d191De18C8275)
    └── ExecutionSuccess
        ├── txHash: 0x328d06921c1675ef29bf0bc6d978bbe893e3dbbf73413e6943e866c93b93576c
        └── payment: 0
```

### CODE REVIEW

**Review by SHAKOTN — APPROVED** [review]
Addresses are correct and corresponding to https://docs.chain.link/chainlink-automation/supported-networks

Register: https://etherscan.io/address/0x02777053d6764996e594c3E88AF1D58D5363a2e6
Registrar:
https://etherscan.io/address/0xDb8e8e2ccb5C033938736aa89Fe4fa1eDfD15a1d

Upkeep will be registered using new `keeper_registrar` contract

File: great_ape_safe/ape_api/chainlink.py [hunk 1]
```diff
@@ -7,12 +7,13 @@ def __init__(self, safe):
         self.safe = safe
 
         # contracts
-        self.relayer = self.safe.contract(
-            r.chainlink.upkeep_registration_requests,
-            interface.IUpkeepRegistrationRequests,
-        )
         self.link = self.safe.contract(r.treasury_tokens.LINK, interface.ILinkToken)
-        self.keeper_registry = self.safe.contract(r.chainlink.keeper_registry)
+        self.keeper_registry = self.safe.contract(
```
Comment (SHAKOTN) [inline great_ape_safe/ape_api/chainlink.py:11]: Not a blocker, just one question. Where is `self.keeper_registry` used? Cannot find any code using this contract

**Review by SHAKOTN — COMMENTED** [review]
Just one comment. Not a blocker

File: great_ape_safe/ape_api/chainlink.py (hunk 1 above)
Comment (gosuto-inzasheru) [reply great_ape_safe/ape_api/chainlink.py:11]: this allows for directly calling the keeper registry once the `Chainlink` class has been initialised, eg in these scripts: https://github.com/Badger-Finance/badger-multisig/search?q=keeper_registry

**Pull request merged**